geps/gep-1651.md

```diff
@@ -0,0 +1,268 @@
+>
+> Multiple scopes Pros:
+>
+> Allows a single Gateway to express multiple networks, saving on needing to attach HTTPRoutes to multiple Gateways for each network scope.
+>
+> Multiple scopes Cons:
+>
+> Complicates the Gateway's purpose. Instead of one Gateway being one set of Listeners, now a Gateway is two sets of listeners that have a totally different scope (and presumably, security context). Personally, I'm also concerned how this will interact with other features like merging and preprovisioning that GEP-1867: Per-Gateway Infrastructure #1868 will allow.
+
+### Adding `reachabiliy` attribute to GatewayClass
+
+See [Prior Art - Multiple Gateways Classes](#multiple-gateway-classes)
+
+## Survey of Prior Art
+
+These alternatives are a survey of existing approaches to support cluster
+local Gateways. Most are implementation specific and are not portable.
+
+### Special annotation/label
+
+Istio let's you specify an annotation `networking.istio.io/service-type` to
+change the underlying Kubernetes Service type to make it a ClusterIP type.
+
+### Re-use of AddressType Hostname
+
+Istio let's you re-use existing Gateway deployments by setting the address
+type to `Hostname` and the value to the Istio ingress Kubernetes Service. If an
+operator configures the Istio deployment to support cluster local traffic a
+Gateway implementation can select it using the `HostName` attribute.
+
+### Multiple Gateway Classes
+
+Some implementations support multiple deployments on a single cluster where each maps to a
+GatewayClass. One of these deployments can be configured to serve cluster local traffic. This is
+sub-optimal because this is implementation specific and the end-user is effectively managing the
+deployments themselves rather than infrastructure being automatically provisioned.
+
+Likewise, infrastructure providers may provide a fixed set of GatewayClasses with unique and fixed
+reachability. Thus GatewayClass name is a viable option to control reachability. But it is not
+flexible enough for implementations that can configure reachability when creating Gateways.
+Otherwise implementations would be forced to create unnecessary GatewayClasses that may cause
+redundant infrastructure to be provisioned.
+
+Additionally, if more attributes are added to GatewayClass to constrain Gateways in some
+form this leads to a combinatorial number of GatewayClassNames. For example, `foo-public` and
+`foo-cluster` are two GatewayClasses surfacing the values of a single attribute `reachability`.
+Let's say we want to enforce address types to just IP then our `gatewayClassName` would be:
+- `foo-public-ipv4`
+- `foo-public-ipv6`
+- `foo-cluster-ipv4`
+- `foo-cluster-ipv6`
+
+As howardjohn mentioned [here](https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1429992160):
+> having the ability to configure things at a higher level seems nice for Gateway, but being able to configure them on a per-Gateway basis remains important.
```

Comment:
Does this need to be an either/or decision? Why not add a reachability field to both Gateway and GatewayClass? The Gateway reachability default value is the GatewayClass's reachability (which is public by default). This way, implementations can support multiple GatewayClass's with different defaults, if they want to, and users can still override by setting reachability in the Gateway, which the implementation may or may not allow, but that is allready addressed in the spec with the accepted status.

Wouldn't this give implementations all the flexibility they need?

Reply:
> Does this need to be an either/or decision? Why not add a reachability field to both Gateway and GatewayClass? 

I'm not opposed to this but I think I'd rather do this incrementally (maybe a future update to this GEP?). Unless implementations can agree on what API changes they want to make to  `GatewayClass`

Reply:
https://github.com/kubernetes-sigs/gateway-api/pull/1868 covers this